### PR TITLE
rgw_sal_motr:[CORTX-32128] Restricted adding tags to delete marker metadata

### PR DESCRIPTION
**Pull request by suryakumar1024**
Problem:
In a versioning Enabled/Suspended bucket, where DeleteMarker exists, GET Object Tagging API without versionId and with DeleteMarker's versionid is returning an empty tagset. Instead, it should return a 'NoSuchKey' error.

Solution
Handled and restricted adding tags to delete marker metadata at get_object method, when some one try to get/ add tags to delete marker it will raise validation error.

Signed-off-by: suryakumar.kumaravelan <suryakumar.kumaravelan@seagate.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by suryakumar1024** [comment]
**Ceph rados response**
[root@ssc-vm-g3-rhev4-3049 ~]# aws s3api put-object-tagging --bucket test-bucket-1 --key object1 --tagging '{"TagSet": [{ "Key": "designation", "Value": "confidential" }]}' --endpoint=$endpoint --version-id "AfqQVzDMAbdtF1rKxAJI2Zbkbpigb7I"

An error occurred (NoSuchBucket) when calling the PutObjectTagging operation: Unknown

**MGW response**
[root@ssc-vm-rhev4-2794 cortx-rgw]# aws s3api put-object-tagging --bucket test-bucket-del --key simpleobj --tagging '{"TagSet": [{ "Key": "designation", "Value": "confidential" }]}' --endpoint=$endpoint --version-id "zWqVTZx6atINOzZQQP8ZsaVhNmnW5QX"

An error occurred (NoSuchKey) when calling the PutObjectTagging operation: Unknown

**Review by sachitanands — APPROVED** [review]
@suryakumar1024  Could you please test this code with latest main branch

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -1528,7 +1528,11 @@ int MotrObject::fetch_obj_entry_and_key(const DoutPrefixProvider* dpp, rgw_bucke
   if (rc < 0) {
       ldpp_dout(dpp, 0) <<__func__<< ": ERROR: failed to get object entry. rc=" << rc << dendl;
       return rc;
-    }
+  }
+  if(ent.is_delete_marker()){
```
Comment (shriya-deshmukh) [inline src/rgw/rgw_sal_motr.cc:1532]: Suggestion: Plz add space between 'if' and '{'.
E.g:
`if (ent.is_delete_marker()) {`

**Review by shriya-deshmukh — APPROVED** [review]

File: src/rgw/rgw_sal_motr.cc (hunk 1 above)
Comment (priyanka25081999) [inline src/rgw/rgw_sal_motr.cc:1532]: cosmetic change:
Please add a space in 'if' condition - line no 1532
`if(ent.is_delete_marker()) {`

**Review by priyanka25081999 — APPROVED** [review]
@suryakumar1024 Please add problem statement and solution in the PR description

**Comment by cdeshmukh** [comment]
@rohit-k-dwivedi plz resolve conflicts

File: src/rgw/rgw_sal_motr.cc (hunk 1 above)
Comment (cdeshmukh) [reply src/rgw/rgw_sal_motr.cc:1532]: yes, lets try to follow the coding standard for rgw now. recent PR 313 had many such corrections, lets avoid any such changes now

Comment (suryakumar1024) [reply src/rgw/rgw_sal_motr.cc]: Addressed in recent patch

**Comment by suryakumar1024** [comment]
Tested with recent main changes:

Postman
`
**URL:** http://ssc-vm-rhev4-2794.colo.seagate.com:8000/test-bucket-del-1/simpleobj?debug=&tagging=
**Body:** 
<?xml version="1.0" encoding="UTF-8"?>
<Tagging xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
   <TagSet>
	  <Tag>
         <Key>objectkey</Key>
         <Value>objectval</Value>
      </Tag>
   </TagSet>
</Tagging>

**Response:**
<?xml version="1.0" encoding="UTF-8"?>
<Error>
    <Code>NoSuchKey</Code>
    <BucketName>test-bucket-del-1</BucketName>
    <HostId></HostId>
</Error>
`

**Comment by suryakumar1024** [comment]
1N deployment success

Custom-CI - > https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6907/console
Deployment-Job -> https://eos-jenkins.colo.seagate.com/job/Cortx-Main/job/rockylinux-8.4/view/Main_Dashboard/job/K8s-1N-deployment/619/


Build image details
```
16:59:23  cortx-docker.colo.seagate.com/seagate/cortx-all:2.0.0-6907-custom-ci
16:59:23  cortx-docker.colo.seagate.com/seagate/cortx-data:2.0.0-6907-custom-ci
16:59:23  cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-6907-custom-ci'
16:59:23  cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-6907-custom-ci
16:59:23  cortx-docker.colo.seagate.com/seagate/cortx-all:2.0.0-6907-custom-ci
16:59:23  cortx-docker.colo.seagate.com/seagate/cortx-data:2.0.0-6907-custom-ci
16:59:23  cortx-docker.colo.seagate.com/seagate/cortx-control:2.0.0-6907-custom-ci
```

**Comment by cdeshmukh** [comment]
>Deployment-Job -> https://eos-jenkins.colo.seagate.com/job/Cortx-Main/job/rockylinux-8.4/view/Main_Dashboard/job/K8s-1N-deployment/619/

again wrong jenkins job used. need to use https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/

**Comment by suryakumar1024** [comment]
> > Deployment-Job -> https://eos-jenkins.colo.seagate.com/job/Cortx-Main/job/rockylinux-8.4/view/Main_Dashboard/job/K8s-1N-deployment/619/
> 
> again wrong jenkins job used. need to use https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/

Okay, Deployed using above mentioned Jenkins job
Custom-CI: https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/6907/
Deployment: https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/7821/

**Pull request merged**